Lib/unittest/mock.py

```diff
@@ -2482,6 +2485,52 @@ def __set__(self, obj, val):
         self(val)
 
 
+class WaitableMock(MagicMock):
+    """
+    A mock that can be used to wait until it was called.
+
+    `event_class` - Class to be used to create event object.
+    Defaults to Threading.Event and can take values like multiprocessing.Event.
+    """
+
+    def __init__(self, *args, event_class=threading.Event, **kwargs):
+        _safe_super(WaitableMock, self).__init__(*args, **kwargs)
+        self._event_class = event_class
+        self._event = event_class()
```

Comment:
The usage self._event vs self._expected_calls which contains events is non obvious. Can add a comment to explain where each event is used and how.